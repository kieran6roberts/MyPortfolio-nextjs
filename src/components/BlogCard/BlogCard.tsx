import Link from "next/link";
import Image from "next/image";

interface BLOG_CARD {
    gridSpan?: string
    imgHeight?: string
}

export default function BlogCard({ gridSpan = "col-span-1", imgHeight = "h-2/5" }: BLOG_CARD) {
    return (
        <article className={`${gridSpan} w-full mx-auto overflow-hidden text-center border border-gray-300 h-hero`}>
            <Link href="/blog">
                <a className="w-full h-full">
                    <div className={`${imgHeight} mb-2 border border-purple-300`}>
                        <Image alt="image"
                        height={150}
                        width={100}
                        objectFit="cover"
                        src="/img.png"
                        />
                    </div>
                    <h3 className="mb-2 text-sm font-bold uppercase">
                        Blog Title
                    </h3>
                    <p className="mb-1 text-xxs">
                        Posted by Kieran Roberts
                    </p>
                    <section className="px-4 py-2">
                        <p className="mb-8 text-xxs">
                            Lorem Ipsum is simply dummy text of the printing and 
                            typesetting industry. Lorem Ipsum has been the industry's 
                            standard dummy text ever since the 1500s. sfvs sdg sd g g gs
                        </p>
                        <footer className="flex items-center justify-between">
                            <div>
                                <span className="mx-4 text-xxs">
                                    Tags
                                </span>
                                <span className="mx-4 text-xxs">
                                    Tags
                                </span>
                            </div>
                            <p className="text-xxs">
                                Posted on <time>time</time>
                            </p>
                        </footer>
                    </section>
                </a>
            </Link>
        </article>
    );
}