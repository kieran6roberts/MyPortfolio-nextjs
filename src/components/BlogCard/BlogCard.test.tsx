import * as React from "react";
import TestRenderer from "react-test-renderer";
import { render, RenderResult } from "@testing-library/react";

import BlogCard from "@/components/BlogCard/BlogCard";

let documentBody: RenderResult;

describe("<BlogCard {...props} />", () => {
    const props = {
        author: "Kieran Roberts",
        description: "This is my first blog post",
        date: "19/01/2021",
        id: "post id",
        previewImage: {
            fileName: "/image.webp"
        },
        subTitle: "subTitle",
        tags: ["test-tag", "blog-tag"],
        title: "Blog Title"
    };

    test("renders", () => {
        documentBody = render(<BlogCard {...props} />);
        
        expect(documentBody.getByRole("article")).toHaveAttribute("id", "post id");
        expect(documentBody.getByAltText("blog preview")).toBeInTheDocument();
        expect(documentBody.getByRole("heading")).toHaveTextContent(/Blog Title/i);
        expect(documentBody.getByText(/kieran roberts/i)).toBeInTheDocument();
        expect(documentBody.getByText(/this is my first blog post/i)).toBeInTheDocument();
    });
    test("renders fallback options ", () => {
        documentBody = render(<BlogCard />);
        
        expect(documentBody.getByRole("heading")).toHaveTextContent("Unable to retrieve title");
        expect(documentBody.getByText("Unable to retrieve date")).toBeInTheDocument();
        expect(documentBody.getByText("Unable to retrieve blog description")).toBeInTheDocument();
        expect(documentBody.getByText("Posted by Kieran Roberts")).toBeInTheDocument();
        expect(documentBody.queryByText("test-tag")).not.toBeInTheDocument();
        expect(documentBody.queryByText("blog-tag")).not.toBeInTheDocument();
    });

    test("render snapshot", () => {
        const tree = TestRenderer
        .create(<BlogCard {...props} />)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});