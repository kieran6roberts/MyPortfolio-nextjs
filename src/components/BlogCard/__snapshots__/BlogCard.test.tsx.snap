// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogCard {...props} /> render snapshot 1`] = `
<article
  className="col-span-1 w-full mx-auto overflow-hidden text-center border border-gray-300 h-hero"
  id="post id"
>
  <a
    className="w-full h-full"
    href="/blog/Blog-Title"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="h-2/5 relative w-full mb-8 border border-purple-300"
    >
      <div
        style={
          Object {
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "left": 0,
            "margin": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <img
          alt="blog preview"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <h3
      className="px-3 mb-6 text-xs font-bold"
    >
      Blog Title
    </h3>
    <p
      className="px-3 mb-3 text-xxs"
    >
      Posted by 
      Kieran Roberts
    </p>
    <section
      className="px-4 py-2"
    >
      <p
        className="null px-3 mb-6 text-xxs"
      >
        This is my first blog post
      </p>
      <footer
        className="flex flex-col items-center justify-between mt-auto md:flex-row"
      >
        <ul
          className="flex flex-wrap justify-center mb-3 md:mb-0"
        >
          <li>
            <span
              className="px-3 py-1 mx-4 my-2 border border-purple-200 text-xxs"
            >
              -
              test-tag
              -
            </span>
          </li>
          <li>
            <span
              className="px-3 py-1 mx-4 my-2 border border-purple-200 text-xxs"
            >
              -
              blog-tag
              -
            </span>
          </li>
        </ul>
        <p
          className="text-xxs"
        >
          Posted on 
          <time>
            19/01/2021
          </time>
        </p>
      </footer>
    </section>
  </a>
</article>
`;
